app/controllers/my-projects/project/hearing/add.js

```diff
@@ -1,4 +1,106 @@
 import Controller from '@ember/controller';
+import EmberObject, { action } from '@ember/object';
+
+class DispositionForAllActions extends EmberObject {
+  publichearinglocation = '';
+
+  dateofpublichearing = '';
+}
 
 export default class MyProjectsProjectHearingAddController extends Controller {
+  // if there is a SINGLE hearing for ALL actions
+  allActions = null;
+
+  modalOpen = false;
+
+  dispositionForAllActions = DispositionForAllActions.create();
+
+  // passed down to date-time-picker and location-input components
+  // set to true when the user clicks on Submit Hearing button
+  // and "submitHearing" action sets it to true
+  // checkIfMissing is for notifying the user which inputs are invalid
+  checkIfMissing = null;
+
+  // to check whether hearing model attributes "location" and "date" are populated
+  // determines certain visual displays based on whether a hearing is submitted
+  hearingSubmitted = false;
+
+  // changes allActions to true or false based on a radio button selection
+  @action
+  setProp(property, newVal) {
+    this.set(property, newVal);
+  }
+
+  @action
+  closeModal() {
+    this.set('modalOpen', false);
+  }
+
+  // checks that the user had input valid values
+  // opens up the modal so the user can review their inputs
+  @action
+  checkInfo() {
```

Comment:
Can we change the naming of this method? Like "onSubmit"?

Reply:
Is `onSubmit` right tho? This action doesn't actually submit, it just validates and opens the modal. Submission happens via another action on a button in the modal.